setup.py

```diff
@@ -41,53 +41,33 @@
         long_description = f.read()
 
     install_requirements = [
+        "vivarium_dependencies[numpy,pandas,pyyaml,scipy,click,tables,loguru,pyarrow,networkx]",
         "vivarium_build_utils>=2.0.1,<3.0.0",
-        "layered_config_tree>=2.1.0",
-        "numpy<2.0.0",
-        "pandas>=2.2.0",
-        "pyyaml>=5.1",
-        "scipy",
-        "click",
-        "tables",
-        "networkx",
-        "loguru",
-        "pyarrow",
+        "layered_config_tree",
         "dill",
-        # Type stubs
-        "pandas-stubs<=2.2.3.250308",
-        "networkx-stubs",
-        "types-PyYAML",
     ]
 
     setup_requires = ["setuptools_scm"]
 
     interactive_requirements = [
-        "IPython",
-        "ipywidgets",
-        "jupyter",
+        "vivarium_dependencies[interactive]",
     ]
 
     test_requirements = [
-        "pytest",
-        "pytest-cov",
-        "pytest-mock",
+        "vivarium_dependencies[pytest]",
         "vivarium_testing_utils",
     ]
 
     lint_requirements = [
-        "black==22.3.0",
+        "vivarium_dependencies[black]",
         "isort",
```

Comment:
Shouldn't this be 
```suggestion
        "vivarium_dependencies[format, mypy]",
```
And if mypy truly needs to be pinned below `1.17.0`, we should have that pin in viv dependencies?

Reply:
I put in vivarium_dependencies the most common pins. If anything deviates from that then it needs to be explicitly defined in the repo. The most common pin for mypy is no pin, and somewhere along the way it was decided that this pin is needed in vivarium so we modified it.